### PR TITLE
[Bug]: Quote index column on db update.

### LINKED ISSUE #14421: [Bug]: Quote index column on db update.
### Expected behavior

Update should work on update in saveIndex()
index column  should be quoted. 

### Actual behavior

You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near 'index = 1 WHERE id = 1' at line 1

https://github.com/pimcore/pimcore/blob/b57e762eb53d931fe96bd4de9afcfc395c05851c/models/Document/Dao.php#L562

### Steps to reproduce

Move a Document in the tree hierarchy.

### PR DESCRIPTION
**Pull request by martineiber**
## Changes in this pull request  
Resolves #14421

### CODE REVIEW

File: models/Document/Dao.php [hunk 1]
```diff
@@ -562,7 +562,7 @@ public function areAllowed(array $columns, User $user): array
     public function saveIndex(int $index): void
     {
         $this->db->update('documents', [
-            'index' => $index,
+            '`index`' => $index,
```
Comment (mattamon) [inline models/Document/Dao.php]: We could use `db->quoteIdentifier` for the quoting.

**Comment by martineiber** [comment]
@mattamon I also removed a non-existing function call `pimcore.elementservice.nodeMoved()`.
This was already removed in https://github.com/pimcore/pimcore/commit/65a67663a08f2414aacc20533b2bdc27a60f3e75 but somehow it got back , but the function dose not exits.

**Comment by mattamon** [comment]
@martineiber nice! Thank you and good catch! 
This function is also called in  pimcore\asset\tree.js Line 474 and pimcore\object\tree.js Line 320

Can we check why it was there again? It seems really strange to me, since it got removed in August already?

**Comment by martineiber** [comment]
@mattamon seams like it was merged back after a merge conflict, resolved in https://github.com/pimcore/pimcore/commit/9fccc1fb2295060a0bde7f29c3355e352a44cd14

**Review by mattamon — APPROVED** [review]

**Pull request merged**